packages/docusaurus-plugin-content-docs/src/versions/files.ts

```diff
@@ -28,27 +28,31 @@ function addPluginIdPrefix(fileOrDir: string, pluginId: string): string {
     : `${pluginId}_${fileOrDir}`;
 }
 
-/** `[siteDir]/community_versioned_docs/version-1.0.0` */
+/** `[siteDir]/[versionedDocsPath]/[community_versioned_docs/version-1.0.0` */
 export function getVersionDocsDirPath(
   siteDir: string,
   pluginId: string,
+  versionedDocsPath: string | undefined,
   versionName: string,
 ): string {
   return path.join(
     siteDir,
+    versionedDocsPath || '',
     addPluginIdPrefix(VERSIONED_DOCS_DIR, pluginId),
     `version-${versionName}`,
   );
 }
 
-/** `[siteDir]/community_versioned_sidebars/version-1.0.0-sidebars.json` */
+/** `[siteDir]/[versionedDocsPath]/community_versioned_sidebars/version-1.0.0-sidebars.json` */
 export function getVersionSidebarsPath(
   siteDir: string,
   pluginId: string,
+  versionedDocsPath: string | undefined,
   versionName: string,
 ): string {
   return path.join(
     siteDir,
+    versionedDocsPath || '',
```

Comment:
I'd rather avoid such default values scattered in multiple places: move them to option normalization process or even better option defaults

Reply:
If versionedDocsPath should remain undefined in option, I'd rather have a `getVersionedDocsPath(siteDir, versionedDocsPathOption)` to apply the fallback in a single place